### PR TITLE
Simplified inheritance of cPluginLua

### PR DESCRIPTION
**Pull request by tigerw**
Nothing breaks!

### CODE REVIEW

**Comment by madmaxoft** [comment]
Why are you so intent on this? Did you decide that you just need to create a merge conflict for me? I say I'll work on the plugin manager, and suddenly everyone starts messing with it, rather needlessly, too.

**Comment by madmaxoft** [comment]
This makes it seem that plugins are allowed to add webadmin tabs to other plugins (`AddWebTab` is available after using `cPluginManager:GetPlugin()` )

**Comment by tigerw** [comment]
My apologies; I would be happy for you to merge your changes and then resolve the conflicts here, if you want.

Since LoadPlugin and GetPlugin are both marked deprecated and unsafe, should I remove them?

**Comment by madmaxoft** [comment]
Those two have already been dealt with in my PR. Have a look there if there's anything you disagree with; if not, we can merge it and then you can have a go at this (although I still don't think this is a good idea)

**Comment by tigerw** [comment]
Tada!

File: src/Bindings/PluginManager.cpp [hunk 1]
```diff
@@ -78,9 +77,9 @@ void cPluginManager::RefreshPluginList(void)
 	// Set all plugins with invalid folders as psNotFound:
 	for (auto & plugin: m_Plugins)
 	{
-		if (std::find(Folders.cbegin(), Folders.cend(), plugin->GetFolderName()) == Folders.end())
+		if (std::find(Folders.cbegin(), Folders.cend(), plugin->GetLocalFolder()) == Folders.end())
```
Comment (madmaxoft) [inline src/Bindings/PluginManager.cpp]: Wrong, `GetLocalFolder` returns the path including the "Plugins/", `Folders` never contains the "Plugins/" prefix.
You need to call `GetFolderName`

File: src/Generating/ChunkGenerator.h [hunk 2]
```diff
@@ -177,7 +177,7 @@ class cChunkGenerator :
 	cGenerator * m_Generator;
 	
 	/** The plugin interface that may modify the generated chunks */
-	cPluginInterface * m_PluginInterface;
+	cPluginLuaInterface * m_PluginInterface;
```
Comment (madmaxoft) [inline src/Generating/ChunkGenerator.h]: I don't think this file needed any of the changes.

**Comment by tigerw** [comment]
Tada!

**Comment by madmaxoft** [comment]
I still don't like this change. It forever breaks the opportunity to have different language plugins, and it exposes some functions to the API that shouldn't be exposed. Is it really worth the few lines you have saved?

**Comment by tigerw** [comment]
There have been requests to add new scripting languages recently, but those have been turned down for one reason or another, and surely the exposed functions can be unexported again.

**Comment by tigerw** [comment]
Merge?

**Comment by madmaxoft** [comment]
I'm gonna say this one more time: I don't like this change and I don't see any benefit in making it.
If there are others who support this change, okay; if not, then I vote DON'T MERGE at all.

**Comment by NiLSPACE** [comment]
I think we need some more voices here, since it makes it really hard to add more scripting languages. @bearbin @worktycho @Seadragon91, @SafwatHalaby @SphinxC0re?

As for me, I don't really like the fact that it would get harder to implement new languages, but on the other hand thus far we rejected every other language. I'm a little 50/50.

**Comment by madmaxoft** [comment]
It's not only about the other languages, this also changes the API in a bit. Suddenly each plugin is capable of calling other plugin's API methods, which hasn't been possible before. Think `OtherPlugin:AddWebTab()` (this one is actually explicitly protected in the manual bindings, but you get the point)

**Comment by bearbin** [comment]
I don't think that I really have the knowledge to say whether this is a good idea. But I think it's quite unlikely that we'll ever have another scripting language, and I don't think there's much of a problem with plugins calling other plugins' API methods - only the plugins that really need it will do it.

**Comment by SafwatHalaby** [comment]
Like @bearbin , I still don't have enough knowledge on this subject to give a worthy opinion. Though I don't imagine we'll have any other scripting languages.

Currently my understanding of MCServer is very narrowly focused, I only learned what I need to learn to get the Pathfinder going, so my knowledge is limited to cChunk and cEntity and its descendants. I'll study the project thoroughly when I have time.

**Comment by tigerw** [comment]
No one's sure. Benefit of the doubt, and revert if problems in the future?

**Comment by worktycho** [comment]
I've got another problem. By allowing plugins to call other plugins API methods you break thread safety. The methods are written to assume that the Plugin Lock is held, but under this code, they could be called from other plugins in which case the lock would not be held.

**Comment by tigerw** [comment]
Guess not then, dam threading. Dam locks. Dam puns.

**Pull request closed without merging**